Rivet/Functions/Checkpoint-Migration.ps1

```diff
@@ -39,29 +36,45 @@ function Checkpoint-Migration
 
     [Rivet.Configuration.Configuration]$settings = Get-RivetConfig -Database $Database -Path $ConfigFilePath -Environment $Environment
 
-    if( -not $OutputPath )
+    foreach( $databaseItem in $settings.Databases )
     {
-        # If no output path put schema.ps1 file in same directory as the rivet.json file.
-        $OutputPath = Join-Path -Path (Split-Path -Path $settings.Path) -ChildPath "schema.ps1"
-    }
+        $OutputPath = Join-Path -Path $databaseItem.MigrationsRoot -ChildPath "schema.ps1"
 
-    if ( (Test-Path -Path $OutputPath) -and -not $Force )
-    {
-        Write-Error "Checkpoint output path ""$($OutputPath)"" already exists. Use the -Force switch to overwrite."
-        return
-    }
+        if ( (Test-Path -Path $OutputPath) -and -not $Force )
+        {
+            Write-Error "Checkpoint output path ""$($OutputPath)"" already exists. Use the -Force switch to overwrite."
+            return
+        }
 
-    $params = @{
-        Database = $Database;
-        Environment = $Environment;
-        ConfigFilePath = $ConfigFilePath;
-    }
+        $query = @"
+        SELECT CONCAT( FORMAT(ID, '00000000000000'), '_', Name) as MigrationFileName
+        FROM rivet.Migrations
+        WHERE ID > 00010101000000
+"@
 
-    Write-Debug "Checkpoint-Migration: Invoke-Rivet -Push on database(s) ($($settings.Databases.Name -join ', '))"
-    Invoke-Rivet -Push @params | Write-Debug
+        Connect-Database -SqlServerName $settings.SqlServerName `
+                            -Database $databaseItem.Name `
+                            -ConnectionTimeout $settings.ConnectionTimeout
 
-    Write-Debug "Checkpoint-Migration: Exporting migration on database(s) ($($settings.Databases.Name -join ', '))"
-    $migration = Export-Migration -SqlServerName $settings.SqlServerName -Database $settings.Databases[0].Name
-    $migration = $migration -join [Environment]::NewLine
-    $migration | Out-File -FilePath $OutputPath
+        $pushedMigrations = Invoke-Query -Query $query
+
+        Disconnect-Database
+
+        if( -not $pushedMigrations -and -not $Force)
+        {
+            Write-Error "There are currently no pushed migrations to checkpoint. Please push migrations before checkpointing."
+            return
+        }
+
+        Write-Debug "Checkpoint-Migration: Exporting migration on database $($databaseItem.Name)"
+        $migration = Export-Migration -SqlServerName $settings.SqlServerName -Database $databaseItem.Name
+        $migration = $migration -join [Environment]::NewLine
+        $migration | Out-File -FilePath $OutputPath

```

Comment:
Default is set to utf8NoBOM but that can be changed. What do you want me to do here?

Reply:
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/out-file?view=powershell-7.2#-encoding

Reply:
Looks like 5.1 is unicode/UTF-16. https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/out-file?view=powershell-5.1#-encoding

Instead of `Out-File`, use `Set-Content`.